test/Libraries/AnalysisTests/SurfaceDataTests.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+using Analysis;
+using Autodesk.DesignScript.Geometry;
+using NUnit.Framework;
+
+namespace AnalysisTests
+{
+    [TestFixture]
+    class SurfaceDataTests : TestServices.GeometricTestBase
+    {
+        /// <summary>
+        /// This test method will execute the BySurfaceAndPoints method from the SurfaceData class passing different parameters
+        /// </summary>
+        [Test, Category("UnitTests")]
+        public void SurfaceAnalysisDataBySurfaceAndPointsTest_Exceptions()
+        {
+            //Act
+            var sad = SurfaceData.BySurfaceAndPoints(TestSurface(),TestUvs());
+
+            //Assert
+            Assert.NotNull(sad);
+            Assert.NotNull(sad.Surface);
+            //The BySurfaceAndPoints method creates a surface without values, then we validate that values is null
+            Assert.IsNull(sad.Values);
+
+            //It will raise an ArgumentNullException when we pass the Surface as null
+            Assert.Throws<ArgumentNullException>(() => SurfaceData.BySurfaceAndPoints(null, TestUvs()));
+
+            //It will raise an ArgumentNullException when we pass the UV values as null
+            Assert.Throws<ArgumentNullException>(() => SurfaceData.BySurfaceAndPoints(TestSurface(), null));
+
+            //It will raise an ArgumentNullException when we pass the UV values as empty IEnumerable
+            Assert.Throws<ArgumentException>(() => SurfaceData.BySurfaceAndPoints(TestSurface(), new UV[0]));
```

Comment:
If it throws `ArgumentNullException`, shouldn't you check for `ArgumentNullException` specifically?

Reply:
@aparajit-pratap 
The SurfaceData.BySurfaceAndPoints method throws a ArgumentNullException when the first parameter or second parameter are null but when the IEnumerable<UV> parameter is empty it throws a ArgumentException, the I think is correct in this way.
Let me know if you agree.